public/app/features/dashboard/dashgrid/DashboardGrid.tsx

```diff
@@ -187,18 +190,81 @@ export class DashboardGrid extends PureComponent<Props> {
     this.updateGridPos(newItem, layout);
   };
 
+  isInView = (panel: PanelModel): boolean => {
+    if (panel.fullscreen || panel.isEditing) {
+      return true;
+    }
+
+    // elem is set *after* the first render
+    const elem = this.panelRef[panel.id.toString()];
+    if (!elem) {
+      // NOTE the gridPos is also not valid until after the first render
+      // since it is passed to the layout engine and made to be valid
+      // for example, you can have Y=0 for everything and it will stack them
+      // down vertically in the second call
+      return false;
+    }
+
+    // TODO? Is there a better way to get this?
```

Comment:
Sadly there is no easy way to get panel position and dimensions from the grid component, https://github.com/STRML/react-grid-layout/issues/506 . which is sad as it has all that info

Reply:
hm.. Maybe we should open a PR to pass it on to children, if that is possible

Reply:
I don't see any clear way to add this in  STRML/react-grid-layout.  The position is calculated in render() and manipulates raw dom style.  We could modify the `onLayoutChange` callback to include Position, but that starts getting a little ugly and would need to recalculate the position anyway.